src/libraries/Microsoft.Extensions.Logging.Abstractions/src/LogValuesFormatter.cs

```diff
@@ -115,15 +115,18 @@ private static int FindIndexOfAny(string format, char[] chars, int startIndex, i
 
         public string Format(object[] values)
         {
+            object[] formattedValues = null;
+
             if (values != null)
             {
+                formattedValues = new object[values.Length];
                 for (int i = 0; i < values.Length; i++)
                 {
-                    values[i] = FormatArgument(values[i]);
+                    formattedValues[i] = FormatArgument(values[i]);
```

Comment:
It looks like FormatArgument will often just return the input, and if that's the case for all inputs, there's no benefit allocating am array. And like this would be better off lazily creating the new array the first time a formatted argument returns a different reference than the input.

Reply:
On top of that, it looks like this method has four call sites, three of which always pass in a brand new array, where it doesn't matter if the method modifies it.  Seems like there should be a separate path for that, if this really is hot.